onnxruntime/core/mlas/inc/mlas.h

```diff
@@ -53,6 +53,12 @@ Module Name:
 #if defined(__VSX__)
 #define MLAS_TARGET_POWER
 #endif
+#if defined(__wasm__)
+#define MLAS_TARGET_WASM
+#if defined(__wasm_simd128__)
```

Comment:
Should add a MLAS_TARGET_WASMSCALAR (or preferably WASM_SCALAR and change WASMSIMD to WASM_SIMD). That would clean up some of the checks in the code.